<?xml version="1.0"?>
<launch>
    <arg name="camera_name_1"           default="zed2i_long" /> <!-- The name you want -->
    <arg name="camera_name_2"           default="zed2i_short" /> <!-- The name you want -->
    <arg name="markerId"        default="422"/>
    <arg name="markerSize"      default="0.043"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
    <arg name="marker_frame1"    default="aruco_marker_frame_long"/>
    <arg name="marker_frame2"    default="aruco_marker_frame_short"/>


    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find vision_pkg)/rviz/zed_aruco_multicam.rviz" output="screen"/>

    <include file="$(find vision_pkg)/launch/multi_zed_merge.launch">
        <arg name="camera_name_1"                     value="$(arg camera_name_1)" />
        <arg name="camera_name_2"                     value="$(arg camera_name_2)" />
    </include>

    <group ns="$(arg camera_name_1)">
        <node pkg="aruco_ros" type="single" name="aruco_single">
            <remap from="/camera_info" to="/zed_multi/zed2i_long/zed_nodelet_front/left/camera_info" />
            <remap from="/image" to="/zed_multi/zed2i_long/zed_nodelet_front/left/image_rect_color" />
            <param name="image_is_rectified" value="True"/>
            <param name="marker_size"        value="$(arg markerSize)"/>
            <param name="marker_id"          value="$(arg markerId)"/>
            <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
            <param name="camera_frame"       value="zed2i_long_left_camera_optical_frame"/>
            <param name="marker_frame"       value="$(arg marker_frame1)" />
            <param name="corner_refinement"  value="$(arg corner_refinement)" />
        </node>
    </group>

    <group ns="$(arg camera_name_2)">
        <node pkg="aruco_ros" type="single" name="aruco_single">
            <remap from="/camera_info" to="/zed_multi/zed2i_short/zed_nodelet_rear/left/camera_info"/>
            <remap from="/image" to="/zed_multi/zed2i_short/zed_nodelet_rear/left/image_rect_color" />
            <param name="image_is_rectified" value="True"/>
            <param name="marker_size"        value="$(arg markerSize)"/>
            <param name="marker_id"          value="$(arg markerId)"/>
            <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
            <param name="camera_frame"       value="zed2i_short_left_camera_optical_frame"/>
            <param name="marker_frame"       value="$(arg marker_frame2)" />
            <param name="corner_refinement"  value="$(arg corner_refinement)" />
        </node>
    </group>
</launch>